File: license-server/src/main/java/de/netzkronehd/license/model/LicenseCheckResult.java
package de.netzkronehd.license.model;

import lombok.*;
import org.springframework.format.annotation.DateTimeFormat;

import java.time.OffsetDateTime;
import java.util.Objects;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@ToString
public class LicenseCheckResult {

    private String licenseKey;
    private String publisher;
    private String notes;
    private boolean valid;
    @DateTimeFormat(
            iso = DateTimeFormat.ISO.DATE_TIME
    )
    private OffsetDateTime validUntil;
    private String signature;

    @Override
    public final boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof LicenseCheckResult that)) return false;
        return valid == that.valid && Objects.equals(licenseKey, that.licenseKey) && Objects.equals(publisher, that.publisher) && Objects.equals(notes, that.notes) && Objects.equals(validUntil, that.validUntil) && Objects.equals(signature, that.signature);
    }

    @Override
    public int hashCode() {
        int result = Objects.hashCode(licenseKey);
        result = 31 * result + Objects.hashCode(publisher);
        result = 31 * result + Objects.hashCode(notes);
        result = 31 * result + Boolean.hashCode(valid);
        result = 31 * result + Objects.hashCode(validUntil);
        result = 31 * result + Objects.hashCode(signature);
        return result;
    }
}

File: license-server-rs/src/model/license_check_result.rs
use std::fmt::{Display, Formatter};

pub struct LicenseCheckResult {
    license_key: String,
    publisher: String,
    notes: String,
    valid: bool,
    valid_until: chrono::DateTime<chrono::FixedOffset>,
    signature: String,
}

impl LicenseCheckResult {
    pub fn new(
        license_key: String,
        publisher: String,
        notes: String,
        valid: bool,
        valid_until: chrono::DateTime<chrono::FixedOffset>,
        signature: String,
    ) -> Self {
        LicenseCheckResult {
            license_key,
            publisher,
            notes,
            valid,
            valid_until,
            signature,
        }
    }

    pub fn license_key(&self) -> &str {
        &self.license_key
    }

    pub fn publisher(&self) -> &str {
        &self.publisher
    }

    pub fn notes(&self) -> &str {
        &self.notes
    }

    pub fn valid(&self) -> bool {
        self.valid
    }

    pub fn valid_until(&self) -> &chrono::DateTime<chrono::FixedOffset> {
        &self.valid_until
    }

    pub fn signature(&self) -> &str {
        &self.signature
    }
}

impl PartialEq for LicenseCheckResult {
    fn eq(&self, other: &Self) -> bool {
        self.license_key == other.license_key
            && self.publisher == other.publisher
            && self.notes == other.notes
            && self.valid == other.valid
            && self.valid_until == other.valid_until
            && self.signature == other.signature
    }
}

impl Display for LicenseCheckResult {
    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
        write!(
            f,
            "LicenseCheckResult {{ license_key: {:?}, publisher: {:?}, notes: {:?}, valid: {:?}, valid_until: {:?}, signature: {:?} }}",
            self.license_key,
            self.publisher,
            self.notes,
            self.valid,
            self.valid_until,
            self.signature
        )
    }
}
